awx/main/tests/functional/task_management/test_scheduler.py

```diff
@@ -214,6 +214,27 @@ def test_single_job_dependencies_project_launch(default_instance_group, job_temp
         TaskManager.start_task.assert_called_once_with(j, default_instance_group, [], instance)
 
 
+@pytest.mark.django_db
+def test_job_not_blocking_project_update(default_instance_group, job_template_factory):
```

Comment:
I'm concerned about using `start_task`  being the test point for this. There's a lot of potential for `start_task` itself to bail out (for things like pre-start checks) and it's not unreasonable to think that at some point it might be refactored in such a way that it masks a job dependency problem.

It might be better to hook into the `TaskManager::is_job_blocked` mechanism here, as well as calling out to `DependencyGraph::is_job_blocked` directly to make sure both code paths are covered independently.

Reply:
@matburt I've updated the test. I now call is_job_blocked from both the TaskManager and DependecyGraph, to check them independently.